snuba/datasets/schemas/join.py

```diff
@@ -65,8 +66,13 @@ class TableJoinNode(TableSource, JoinNode):
     It can be a table or a view.
     """
 
-    def __init__(self, table_name: str, columns: ColumnSet, alias: str) -> None:
-        super().__init__(table_name, columns)
+    def __init__(self,
+        table_name: str,
+        columns: ColumnSet,
+        mandatory_conditions: Optional[Sequence[Condition]],
```

Comment:
Should this also have a default of `None`? Is this just because of the parameter order? I guess this relates to the comment below a bit too…

Reply:
indeed this is just for parameters order. In order to be consistent with the order of the parent class this cannot have a default value, unless I impose that every parameter after columns is not positional.
In the end it seems better to me having to provide a parameter that has a trivial default value than breaking the consistency with the parent class.

Reply:
Makes sense, though I think that this might be an indication that `TableJoinNode` should actually take a `TableSource` as a parameter, rather than subclassing it.